tests/components/openrgb/test_light.py

```diff
@@ -0,0 +1,663 @@
+    ATTR_RGB_COLOR,
+    DOMAIN as LIGHT_DOMAIN,
+    EFFECT_OFF,
+)
+from homeassistant.components.openrgb.const import (
+    DEFAULT_BRIGHTNESS,
+    DEFAULT_COLOR,
+    DOMAIN,
+    OFF_COLOR,
+    OpenRGBMode,
+)
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    SERVICE_TURN_OFF,
+    SERVICE_TURN_ON,
+    STATE_OFF,
+    STATE_ON,
+    STATE_UNAVAILABLE,
+    Platform,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers import device_registry as dr, entity_registry as er
+
+from tests.common import MockConfigEntry, snapshot_platform
+
+
+@pytest.fixture(autouse=True)
+def light_only() -> Generator[None]:
+    """Enable only the light platform."""
+    with patch(
+        "homeassistant.components.openrgb.PLATFORMS",
+        [Platform.LIGHT],
+    ):
+        yield
+
+
+# Test basic entity setup and configuration
+@pytest.mark.usefixtures("init_integration")
+async def test_entities(
+    hass: HomeAssistant,
+    snapshot: SnapshotAssertion,
+    entity_registry: er.EntityRegistry,
+    device_registry: dr.DeviceRegistry,
+    mock_config_entry: MockConfigEntry,
+) -> None:
+    """Test the light entities."""
+    await snapshot_platform(hass, entity_registry, snapshot, mock_config_entry.entry_id)
+
+    # Ensure entities are correctly assigned to device
+    device_entry = device_registry.async_get_device(
+        identifiers={
+            (
+                DOMAIN,
+                f"{mock_config_entry.entry_id}||LEDSTRIP||Test Vendor||Test LED Strip||TEST123||Test Location",
+            )
+        }
+    )
+    assert device_entry
+    entity_entries = er.async_entries_for_config_entry(
+        entity_registry, mock_config_entry.entry_id
+    )
+    # Filter out the server device
+    entity_entries = [e for e in entity_entries if e.device_id == device_entry.id]
+    assert len(entity_entries) == 1
+    assert entity_entries[0].device_id == device_entry.id
+
+
+@pytest.mark.usefixtures("mock_openrgb_client")
+async def test_light_with_black_leds(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+    mock_openrgb_device: MagicMock,
+) -> None:
+    """Test light state when all LEDs are black (off by color)."""
+    # Set all LEDs to black
+    mock_openrgb_device.colors = [RGBColor(*OFF_COLOR), RGBColor(*OFF_COLOR)]
+    mock_openrgb_device.active_mode = 0  # Direct mode (supports colors)
+
+    mock_config_entry.add_to_hass(hass)
+    await hass.config_entries.async_setup(mock_config_entry.entry_id)
+    await hass.async_block_till_done()
+
+    assert mock_config_entry.state is ConfigEntryState.LOADED
+
+    # Verify light is off by color
+    state = hass.states.get("light.test_rgb_device")
+    assert state
+    assert state.state == STATE_OFF
+    assert state.attributes.get("rgb_color") is None
+    assert state.attributes.get("brightness") is None
```

Comment:
bonus points if we can use `ATTR_*` constants for the keys we get from the state attributes